src/app/left-menu/main/left-menu.component.ts

```diff
@@ -52,23 +53,112 @@ export class LeftMenuComponent implements OnDestroy, OnInit {
-        this.filteredMenuData = this.menuData.filter((value) => {
-            const name = value.name.toLowerCase();
-            const searchString = name + ', ' + value.equivalents;
-            return searchString.toLowerCase().indexOf(keyword) !== -1;
-        });
+        const keyword = this.getRealName(this.searchKeyword.toLowerCase());
+
+        if (keyword.length < 3) {
+            this.filteredData = [];
+            this.filteredData[0] = this.workArea;
+        }
+
+        if (keyword.length >= 3) {
+            this.filteredData = this.deepClone(this.menuData);
+
+            this.filteredData.forEach((data) => {
+                if (data.groups) {
+                    data.groups.forEach((group) => {
+                        if (group.items) {
+                            group.items = group.items.filter((value) => {
+                                const searchString = this.getRealName(value.Name);
+                                if(searchString.toLowerCase() === keyword) {
+                                    return true;
+                                } else if(searchString.toLowerCase().slice(0, keyword.length - searchString.length) === keyword) {
+                                    return true;
+                                } else {
+                                    return false;
+                                }
+                            });
+                        }
+                    });
+                } else {
+                    data.items = data.items.filter((value) => {
+                        const searchString = this.getRealName(value.Name);
+                        if (searchString.toLowerCase() === keyword) {
+                            return true;
+                        } else if (searchString.toLowerCase().slice(0, keyword.length - searchString.length) === keyword) {
+                            return true;
+                        } else {
+                            return false;
+                        }
+                    });
+                }
+            });
+
+            for(let i = 0; i < this.filteredData.length; i++) {
+                if(this.filteredData[i].items) {
+                    if(this.filteredData[i].items.length === 0) {
+                        this.filteredData.splice(i, 1);
+                        i--;
+                    }
+                } else {
+                    for(let k = 0; k < this.filteredData[i].groups.length; k++) {
+                        if(this.filteredData[i].groups[k].items) {
+                            if(this.filteredData[i].groups[k].items.length === 0) {
+                                this.filteredData[i].groups.splice(k, 1);
+                                k--;
+                            }
+                        }
+                    }
+                    if(this.filteredData[i].groups.length === 0 || (this.filteredData[i].groups.length === 4 && this.filteredData[i].name === 'Editors')) {
```

Comment:
Hm. Editors? What about Buttons? (Button have groups too)

Reply:
Because "Editors" contains emty objects without "items" or "groups". For example "Editors"  contain "'Autocomplete"